From c8f0e86b1382b7d2ffbab5f2805e1d069a39cd7e Mon Sep 17 00:00:00 2001
From: Rogerio Guerra Borin <rogerio.borin@toradex.com>
Date: Wed, 9 Oct 2024 20:23:16 -0300
Subject: [PATCH] toradex: add helper to detect closed k3 device (am62)

Upstream-Status: Inappropriate [Torizon OS specific]

Signed-off-by: Rogerio Guerra Borin <rogerio.borin@toradex.com>
---
 arch/arm/mach-k3/common.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/mach-k3/common.c b/arch/arm/mach-k3/common.c
index 1ddb2917307..a3d542ae29a 100644
--- a/arch/arm/mach-k3/common.c
+++ b/arch/arm/mach-k3/common.c
@@ -587,6 +587,12 @@ static const char *get_device_type_name(void)
 	}
 }
 
+int tdx_secboot_k3_dev_is_closed(void)
+{
+	/* Device is closed (security enforced (SE) state). */
+	return (get_device_type() == K3_DEVICE_TYPE_HS_SE);
+}
+
 int print_cpuinfo(void)
 {
 	struct udevice *soc;
-- 
2.34.1

